.github/workflows/jobs-create-env-mock-data.yml

```diff
@@ -0,0 +1,112 @@
+name: Mock up data for environment
+
+on:
+  workflow_call:
+    inputs:
+      environment_name:
+        type: string
+        required: true
+    secrets:
+      API_key:
+        required: true
+
+jobs:
+  create_mock_data:
+    env:
+      baseAKSURL: https://byk-${{ inputs.environment_name }}-aks-ingress.westeurope.cloudapp.azure.com
+      centopsAdminInstitutionRoute: centops/admin/institutions
+      centopsParticipantsRoute: centops/admin/participants
+      dmrRoute: dmr/messages
+      mockBotRoute: mock-bot/dmr-api
+      mockClassifierRoute: mock-classifier/dmr-api/messages
+      institutionName: mock-institution
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - name: Teardown mock data
+        run: |
+          institutions=$(curl \
+          -X GET \
+          -H "Content-Type: application/json" \
+          -H "X-Api-Key: ${{ secrets.API_key }}" \
+          ${{ env.baseAKSURL }}/${{ env.centopsAdminInstitutionRoute }} \
+          )
+
+          institutionsByName=""
+
+          if [ ${#institutions} -gt 2 ] 
+          then
+            institutionsByName=$( echo "$institutions" | jq -c '[.[] | select(.name | contains("${{ env.institutionName }}"))]' )
+          fi
+
+          if [ ${#institutionsByName} -gt 2 ] 
+          then
+          #Participants            
+            participants=$(curl \
+            -X GET \
+            -H "Content-Type: application/json" \
+            -H "X-Api-Key: ${{ secrets.API_key }}" \
+            ${{ env.baseAKSURL }}/${{ env.centopsParticipantsRoute }} \
+            )
+
+            if [ ${#participants} -gt 2 ] 
+              then
+                institutionIdDelete=$( echo "$institutionsByName" | jq -c '[.[].id]' )      
+                participantsToDelete=$(echo $participants | jq -r '[.[] | select(.institutionId == ('"$institutionIdDelete"'|unique[]))]')
+
+                echo $participantsToDelete | jq -r '.[].id' | while read id ; do 
+                  url=$(echo "${{ env.baseAKSURL }}/${{ env.centopsParticipantsRoute }}/${id}" | xargs)
+                  curl \
+                  -X DELETE \
+                  -H "X-Api-Key: ${{ secrets.API_key }}" \
+                  $url
+                done
+              fi
+
+          #institutions
+            echo $institutionsByName | jq -r '.[].id' | while read id ; do 
+              url=$(echo "${{ env.baseAKSURL }}/${{ env.centopsAdminInstitutionRoute }}/${id}" | xargs)
+              curl \
+              -X DELETE \
+              -H "X-Api-Key: ${{ secrets.API_key }}" \
+              $url
+            done
+          fi  
+
+      - name: Create mock data
+        run: |
+          createInstitutionResponse=$(curl \
```

Comment:
I do agree this would be a good approach but there was a couple of reasons why I decided to keep the script it in the action only:

1. Passing the variables and secrets to the bash commands is easier 
2. This script is very light weight, would expect it to be removed or changed frequently in the future so keeping it as one file make it more mobile

Happy to move it to a bash script though if you disagree though :) should not be too much extra work as well